packages/firestore/src/util/misc.ts

```diff
@@ -28,8 +29,17 @@ export class AutoId {
     const chars =
       'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
     let autoId = '';
-    for (let i = 0; i < 20; i++) {
-      autoId += chars.charAt(Math.floor(Math.random() * chars.length));
+    while (autoId.length < 20) {
```

Comment:
Consider re-writing this loop as `while (true)` and then breaking out of the loop with a `break` statement immediately after appending the 20th character to `autoId`. This would avoid hardcoding `20` in two different places.

Reply:
Rewrote the section to reduce hardcoding numbers.